imagenet/main.py

```diff
@@ -48,18 +49,66 @@
                     help='evaluate model on validation set')
 parser.add_argument('--pretrained', dest='pretrained', action='store_true',
                     help='use pre-trained model')
+parser.add_argument('--finetune', dest='finetune', action='store_true',
+                    help='fine tune pre-trained model')
 
 best_prec1 = 0
 
+class FineTuneModel(nn.Module):
+    def __init__(self, original_model, arch, num_classes):
+        super(FineTuneModel, self).__init__()
+
+        if arch.startswith('alexnet') :
+            self.features = original_model.features
+            self.classifier = nn.Sequential(
```

Comment:
This will recreate all the fully-connected layers of your network, which might be too much if your training data is small.
One usually only replace the last fully-connected layer for finetuning.